Index: trunk/usr/share/linuxmuster-portfolio/inc/auth/ad.class.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/inc/auth/ad.class.php	2011-03-17 19:09:06.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/inc/auth/ad.class.php	2011-03-17 19:20:46.000000000 +0100
@@ -30,6 +30,9 @@
  *   // add a list of comma separated ldap contact fields.
  *   $conf['auth']['ad']['additional'] = 'field1,field2';
  *
+ *   // patched for openschulportfolio
+ *   $conf['auth']['ad']['groupdelprefix']     = 'g_projekt_'
+ *
  *  @license GPL 2 (http://www.gnu.org/licenses/gpl.html)
  *  @author  James Van Lommel <jamesvl@gmail.com>
  *  @link    http://www.nosq.com/blog/2005/08/ldap-activedirectory-and-dokuwiki/
@@ -187,6 +190,14 @@
      * @author  James Van Lommel (jamesvl@gmail.com)
      */
     function cleanGroup($name) {
+        if (isset($this->cnf['groupdelprefix'])) {
+            $prefixes = explode("_", $this->cnf['groupdelprefix']);
+            $sname = $name;
+            foreach ($prefixes as $prefix) {
+                $prefix .= "_";
+                $sname = preg_replace("/^$prefix/", '',$sname);
+            }
+        }
         $sName = str_replace('\\', '', $name);
         $sName = str_replace('#', '', $sName);
         $sName = preg_replace('[\s]', '_', $sName);
