Index: openschulportfolio/usr/share/linuxmuster-portfolio/lib/plugins/authldap/auth.php
===================================================================
--- openschulportfolio.orig/usr/share/linuxmuster-portfolio/lib/plugins/authldap/auth.php	2013-04-20 17:45:46.010649077 +0200
+++ openschulportfolio/usr/share/linuxmuster-portfolio/lib/plugins/authldap/auth.php	2013-04-20 18:05:21.746590916 +0200
@@ -202,6 +202,23 @@
         $info['name'] = $user_result['cn'][0];
         $info['grps'] = array();
 
+        # begin OSP modification
+        # Substitutions for mailaddresses for openSchulportfolio
+        # 2013-04-20 Frank Schiebel <frank@linuxmuster.net>
+        if ( file_exists(DOKU_CONF."ldap_maildomains.conf") ){
+            $mailsubs = confToHash(DOKU_CONF."ldap_maildomains.conf");
+            foreach ($mailsubs as $old => $new) {
+                $info['mail'] = preg_replace("/\@$old/","@$new",$info['mail']);
+            }
+        }
+        if ( file_exists(DOKU_CONF."ldap_mailaccounts.conf") ){
+            $mailsubs = confToHash(DOKU_CONF."ldap_mailaccounts.conf");
+            foreach ($mailsubs as $old => $new) {
+                $info['mail'] = preg_replace("/$old\@/","$new@",$info['mail']);
+            }
+        }
+        # end OSP modification
+
         // overwrite if other attribs are specified.
         if(is_array($this->getConf('mapping'))) {
             foreach($this->getConf('mapping') as $localkey => $key) {
@@ -242,7 +259,16 @@
             if(is_array($result)) foreach($result as $grp) {
                 if(!empty($grp[$this->getConf('groupkey')][0])) {
                     $this->_debug('LDAP usergroup: '.htmlspecialchars($grp[$this->getConf('groupkey')][0]), 0, __LINE__, __FILE__);
-                    $info['grps'][] = $grp[$this->getConf('groupkey')][0];
+                    # begin OSP modification
+                    # Enabling groupdelprefix option for integration in linuxmuster.net
+                    # 2013-04-20 Frank Schiebel <frank@linuxmuster.net>
+                    $prefix = $this->getConf('groupdelprefix');
+                    if ($prefix != "") {
+                        $info['grps'][] = preg_replace("/^$prefix/", '', $grp[$this->getConf('groupkey')][0]);
+                    } else {
+                        $info['grps'][] = $grp[$this->getConf('groupkey')][0];
+                    }
+                    # end OSP modification
                 }
             }
         }
