Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/syntax.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/syntax.php	2010-11-14 17:18:37.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/syntax.php	2010-11-14 17:19:50.000000000 +0100
@@ -202,8 +202,49 @@
                 }
               }
               
-              $renderer->doc .= "<table width='100%' border='0' ><tr>";
-              $renderer->doc .= "<td width='60%' valign='top'>";
+              $renderer->doc .= '<div id="bookcreator_exportform">'; 
+              
+              //Esportazione PDF
+              $renderer->doc .= '<form class="button" method="get" action="'.wl($ID).'" accept-charset="'.$lang['encoding'].'">';
+              $renderer->doc .= "<fieldset style=\"text-align:left;\"><legend><b>".$this->getLang('bookcreator_export')."</b></legend>";
+              $renderer->doc .= $this->getLang('bookcreator_title')." ";
+              $renderer->doc .= '<input type="text" class="edit" value="'.$title.'" name="pdfbook_title" size="40" />';
+              $renderer->doc .= '<select name="do" size="1">';
+              $renderer->doc .= '<option value="export_html" selected="selected">'.$this->getLang('bookcreator_exportprint').'</option>';
+
+              if (file_exists(DOKU_PLUGIN."text/renderer.php") && !plugin_isdisabled("text")) {
+                $renderer->doc .= '<option value="export_text">'.$this->getLang('bookcreator_exporttext').'</option>';
+              }
+
+              if (file_exists(DOKU_PLUGIN."dw2pdf/action.php") && !plugin_isdisabled("dw2pdf")) {
+                $renderer->doc .= '<option value="export_pdfbook" selected="selected">'.$this->getLang('bookcreator_exportpdf').'</option>';
+              }
+              
+              $renderer->doc .= '</select>';
+              $renderer->doc .= '<input type="submit" value="'.$this->getLang('bookcreator_create').'" class="button"/>
+                  <input type="hidden" name="id" value="'.$ID.'" />';
+              $renderer->doc .= '</fieldset>';
+              $renderer->doc .= formSecurityToken(false);
+              $renderer->doc .= '</form></div>';
+              //Esportazione PDF
+              
+              
+              $renderer->doc .= '<div id="bookcreator_saveselform">';
+              if ($this->usercansave) {
+                //Salva selezione
+                $renderer->doc .= '<form class="button" method="post" action="'.wl($ID).'" accept-charset="'.$lang['encoding'].'">';
+                $renderer->doc .= "<fieldset style=\"text-align:left;\"><legend><b>".$this->getLang('bookcreator_saveselection')."</b></legend>";
+                $renderer->doc .= '<input type="text" class="edit" value="'.$title.'" name="bookcreator_title" />';
+                $renderer->doc .= '<input type="submit" value="'.$this->getLang('bookcreator_save').'" class="button"/>';
+                $renderer->doc .= '<input type="hidden" name="task" value="save" />
+                    <input type="hidden" name="id" value="'.$ID.'" />';
+                $renderer->doc .= '</fieldset>';
+                $renderer->doc .= formSecurityToken(false);
+                $renderer->doc .= '</form>';
+                //Salva selezione
+              }
+              
+              $renderer->doc .= '</div>';
 
               // Pagine selezionate
               for ($n = 1; $n <= $i; $n++) {            
@@ -258,53 +299,8 @@
               $renderer->doc .= '</div>';
               // azzera selezione
 
-              $renderer->doc .= "</td>";
-              $renderer->doc .= "<td width='40%' valign='top' >";
 
-              $renderer->doc .= "<div align='center'>"; 
-              
-              //Esportazione PDF
-              $renderer->doc .= '<form class="button" method="get" action="'.wl($ID).'" accept-charset="'.$lang['encoding'].'">';
-              $renderer->doc .= "<fieldset style=\"text-align:left;\"><legend><b>".$this->getLang('bookcreator_export')."</b></legend>";
-              $renderer->doc .= $this->getLang('bookcreator_title')." ";
-              $renderer->doc .= '<input type="text" class="edit" value="'.$title.'" name="pdfbook_title" size="40" />';
-              $renderer->doc .= '<select name="do" size="1">';
-              $renderer->doc .= '<option value="export_html" selected="selected">'.$this->getLang('bookcreator_exportprint').'</option>';
-
-              if (file_exists(DOKU_PLUGIN."text/renderer.php") && !plugin_isdisabled("text")) {
-                $renderer->doc .= '<option value="export_text">'.$this->getLang('bookcreator_exporttext').'</option>';
-              }
-
-              if (file_exists(DOKU_PLUGIN."dw2pdf/action.php") && !plugin_isdisabled("dw2pdf")) {
-                $renderer->doc .= '<option value="export_pdfbook" selected="selected">'.$this->getLang('bookcreator_exportpdf').'</option>';
-              }
-              
-              $renderer->doc .= '</select>';
-              $renderer->doc .= '<input type="submit" value="'.$this->getLang('bookcreator_create').'" class="button"/>
-                  <input type="hidden" name="id" value="'.$ID.'" />';
-              $renderer->doc .= '</fieldset>';
-              $renderer->doc .= formSecurityToken(false);
-              $renderer->doc .= '</form>';
-              //Esportazione PDF
-              
-              if ($this->usercansave) {
-                //Salva selezione
-                $renderer->doc .= '<form class="button" method="post" action="'.wl($ID).'" accept-charset="'.$lang['encoding'].'">';
-                $renderer->doc .= "<fieldset style=\"text-align:left;\"><legend><b>".$this->getLang('bookcreator_saveselection')."</b></legend>";
-                $renderer->doc .= '<input type="text" class="edit" value="'.$title.'" name="bookcreator_title" />';
-                $renderer->doc .= '<input type="submit" value="'.$this->getLang('bookcreator_save').'" class="button"/>';
-                $renderer->doc .= '<input type="hidden" name="task" value="save" />
-                    <input type="hidden" name="id" value="'.$ID.'" />';
-                $renderer->doc .= '</fieldset>';
-                $renderer->doc .= formSecurityToken(false);
-                $renderer->doc .= '</form>';
-                //Salva selezione
-              }
-              
-              $renderer->doc .= '</div>';
 
-              $renderer->doc .= "</tr></td>";
-              $renderer->doc .= "</table>";
 
             } else {
               $renderer->doc .= $this->getLang('bookcreator_nocookies');
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/style.css
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/style.css	2010-11-14 17:19:57.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/bookcreator/style.css	2010-11-14 17:21:08.000000000 +0100
@@ -44,4 +44,15 @@
   
 }
 
+div#bookcreator_exportform {
+ width: 45%;
+ float: left;
+}
+
+div#bookcreator_saveselform {
+ width: 45%;
+ float: right;
+}
+
+
 
