Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/dw2pdf/action.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/dw2pdf/action.php	2010-06-15 19:43:39.000000000 +0200
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/dw2pdf/action.php	2010-06-15 19:45:13.000000000 +0200
@@ -48,6 +48,15 @@
         $event->preventDefault();
 
 	$idparam = $ID;
+
+ 	//check read permission (Frank Schiebel <frank@linuxmuster.net>)
+	$perm = auth_quickaclcheck(trim(strtolower($idparam),":"));
+	if ($perm < AUTH_READ) {
+		$relogin = wl($idparam, false, true, "&");
+		Header("Location: $relogin");
+		exit(); 
+	}
+
         if ($REV != 0) {  $idparam = $idparam."&rev=".$REV; };
 
         $pos = strrpos(utf8_decode($ID), ':');
