Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/note/syntax.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/note/syntax.php	2011-03-10 10:54:09.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/note/syntax.php	2011-03-10 10:57:00.000000000 +0100
@@ -106,7 +106,7 @@
 
           switch ($state) {
             case DOKU_LEXER_ENTER :
-              $renderer->doc .= '<p><div class="'.$data.'">';
+              $renderer->doc .= "</p>\n".'<div class="'.$data.'">'."\n<p>";
               break;
   
             case DOKU_LEXER_UNMATCHED :
@@ -114,7 +114,7 @@
               break;
   
             case DOKU_LEXER_EXIT :
-              $renderer->doc .= "\n</div></p>";
+              $renderer->doc .= "</p>\n</div>\n<p>";
               break;
           }
           return true;
