Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php	2010-12-01 03:00:17.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php	2010-12-01 03:12:23.000000000 +0100
@@ -54,6 +54,7 @@
     function connectTo($mode) {
         $this->Lexer->addSpecialPattern('\{\{filename>.+?\}\}',$mode,'plugin_filelist');
         $this->Lexer->addSpecialPattern('\{\{filelist>.+?\}\}',$mode,'plugin_filelist');
+        $this->Lexer->addSpecialPattern('\{\{simplefilelist>.+?\}\}',$mode,'plugin_filelist');
     }
 
     /**
@@ -97,6 +98,14 @@
             $params[trim($name)] = trim($value);
         }
 
+        if ($type == 'simplefilelist' ) {
+            $params['style']="table";
+            $params['tableheader']=1;
+            $params['tableshowdate']=1;
+            $params['tableshowsize']=1;
+            $type = 'filelist';
+        }
+
         // recursive filelistings are not supported for the filename command
         if ($type == 'filename') {
             $params['recursive'] = 0;
@@ -122,6 +131,18 @@
                 $result = $this->_filter_out_directories($result);
             }
 
+            // link to mm
+            // we don't want to use $conf['media'] here as that path has symlinks resolved
+            if (!$params['direct']) {
+                // if media path is not absolute, precede it with the current namespace
+                if ($pattern[0] != ':') {
+                    $pattern = ':'.getNS($ID) . ':' . $pattern;
+                }
+                if ($params['style'] == "table" && auth_quickaclcheck($pattern)>=AUTH_UPLOAD ) {
+                    $renderer->doc .=  "<a href=\"".DOKU_BASE."lib/exe/mediamanager.php?ns=".$pattern."\" rel=\"nofollow\" class=\"fleditfiles\" target=\"mediamanger\">Dateien bearbeiten</a>";
+                }
+            }
+
             // if we got nothing back, display a message
             if ($result == DOKU_PLUGIN_FILELIST_NOMATCH) {
                 $renderer->doc .= '[n/a: ' . $this->getLang('error_nomatch') . ']';
@@ -136,6 +157,9 @@
                 $result['files'] = array_slice($result['files'], $params['offset'], $params['limit']);
             }
 
+
+
+
             switch ($type) {
 
                 case 'filename':
@@ -200,7 +224,11 @@
         $link['more']   = '';
         $link['class']  = 'media';
         if (!$params['direct']) {
-            $link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));
+        $link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));
+            # Remove multiple slashes / or : (June 06 1010, Frank Schiebel frank@linuxmuster.net)
+        $link['url'] = preg_replace("|/+|", '/', $link['url']);
+        $link['url'] = preg_replace("|:+|", ':', $link['url']);
+        $link['url'] = preg_replace("|/:|", '/', $link['url']);
         } else {
             $link['url'] = $webdir.substr($filepath, strlen($basedir));
         }
