Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php	2010-12-15 12:23:55.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/syntax.php	2010-12-15 18:57:04.000000000 +0100
@@ -54,6 +54,7 @@
     function connectTo($mode) {
         $this->Lexer->addSpecialPattern('\{\{filename>.+?\}\}',$mode,'plugin_filelist');
         $this->Lexer->addSpecialPattern('\{\{filelist>.+?\}\}',$mode,'plugin_filelist');
+        $this->Lexer->addSpecialPattern('\{\{simplefilelist>.+?\}\}',$mode,'plugin_filelist');
     }
 
     /**
@@ -84,19 +85,36 @@
             'index' => 0,
             'limit' => 0,
             'offset' => 0,
+            'toolbar' => 0,
             'style' => 'list',
             'tableheader' => 0,
             'tableshowsize' => 0,
             'tableshowdate' => 0,
             'direct' => 0,
             'recursive' => 0,
+            'groupbyext' => 0,
             'titlefile' => '_title.txt',
         );
+
         foreach($flags as $flag) {
             list($name, $value) = split('=', $flag);
             $params[trim($name)] = trim($value);
         }
 
+        if ($type == 'simplefilelist' ) {
+            $params['style']="table";
+            $type = 'filelist';
+
+            // get defs
+            $simplefilelist_defaults = explode(",", $this->getConf('simplefilelist_defaults'));
+
+            if (in_array("tableheader", $simplefilelist_defaults)) $params['tableheader']=1;
+            if (in_array("toolbar", $simplefilelist_defaults)) $params['toolbar']=1;
+            if (in_array("tableshowdate", $simplefilelist_defaults)) $params['tableshowdate']=1;
+            if (in_array("tableshowsize", $simplefilelist_defaults)) $params['tableshowsize']=1;
+            if (in_array("groupbyext", $simplefilelist_defaults)) $params['groupbyext']=1;
+        }
+
         // recursive filelistings are not supported for the filename command
         if ($type == 'filename') {
             $params['recursive'] = 0;
@@ -122,8 +140,26 @@
                 $result = $this->_filter_out_directories($result);
             }
 
+            // build toolbar
+            $listtoolbar = "";
+            // we don't want to use $conf['media'] here as that path has symlinks resolved
+            if (!$params['direct']) {
+                // if media path is not absolute, precede it with the current namespace
+                if ($pattern[0] != ':') {
+                    $pattern = ':'.getNS($ID) . ':' . $pattern;
+                }
+                $pattern = preg_replace('/(^.*:)\*.*$/', '$1', $pattern);
+                if ($params['toolbar'] && auth_quickaclcheck($pattern)>=AUTH_UPLOAD ) {
+                    $listtoolbar .= "<div class=\"filelist-toolbar\">".DOKU_LF;
+                    $listtoolbar .=  "<a href=\"".DOKU_BASE."lib/exe/mediamanager.php?ns=".$pattern."\" rel=\"nofollow\" class=\"filelist-editfiles\" target=\"mediamanger\">".$this->getLang('edit_files')."</a>";
+                    $listtoolbar .=  "<a href=\"javascript:window.location.reload()\" class=\"filelist-refresh\">".$this->getLang('refresh_filelist') ."</a>";
+                    $listtoolbar .= "</div>".DOKU_LF;
+                }
+            }
+
             // if we got nothing back, display a message
             if ($result == DOKU_PLUGIN_FILELIST_NOMATCH) {
+                $renderer->doc .= $listtoolbar;
                 $renderer->doc .= '[n/a: ' . $this->getLang('error_nomatch') . ']';
                 return true;
             } else if ($result == DOKU_PLUGIN_FILELIST_OUTSIDEJAIL) {
@@ -136,6 +172,9 @@
                 $result['files'] = array_slice($result['files'], $params['offset'], $params['limit']);
             }
 
+
+
+
             switch ($type) {
 
                 case 'filename':
@@ -154,12 +193,14 @@
                         case 'list':
                         case 'olist':
                             $renderer->doc .= '<div class="filelist-plugin">'.DOKU_LF;
+                            $renderer->doc .= $listtoolbar.DOKU_LF;
                             $this->_render_list($result, $params, $renderer);
                             $renderer->doc .= '</div>'.DOKU_LF;
                             break;
 
                         case 'table':
                             $renderer->doc .= '<div class="filelist-plugin">'.DOKU_LF;
+                            $renderer->doc .= $listtoolbar.DOKU_LF;
                             $this->_render_table($result, $params, $renderer);
                             $renderer->doc .= '</div>'.DOKU_LF;
                             break;
@@ -200,7 +241,11 @@
         $link['more']   = '';
         $link['class']  = 'media';
         if (!$params['direct']) {
-            $link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));
+        $link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));
+        # Remove multiple slashes / or : (June 06 1010, Frank Schiebel frank@linuxmuster.net)
+        $link['url'] = preg_replace("|/+|", '/', $link['url']);
+        $link['url'] = preg_replace("|:+|", ':', $link['url']);
+        $link['url'] = preg_replace("|/:|", '/', $link['url']);
         } else {
             $link['url'] = $webdir.substr($filepath, strlen($basedir));
         }
@@ -472,7 +517,8 @@
             $callback = array($this, '_compare_names');
             if ($params['order'] == 'desc') $reverseflag = true;
         }
-        $this->_sort_filelist($result['files'], $callback, $reverseflag);
+        if ($params['groupbyext'] == '1') $groupbyextflag = true;
+        $this->_sort_filelist($result['files'], $callback, $reverseflag, $groupbyextflag);
 
         // return the list
         if (count($result['files']) > 0)
@@ -490,20 +536,39 @@
      * @param $reverse true if the result is to be reversed
      * @return the sorted tree
      */
-    function _sort_filelist(&$files, $callback, $reverse) {
-        // sort subtrees
+    function _sort_filelist(&$files, $callback, $reverse, $groupbyextflag=false) {
+
         for ($i = 0; $i < count($files); $i++) {
             if ($files[$i]['children'] !== false) {
                 $children = $files[$i]['children'];
-                $this->_sort_filelist($children, $callback, $reverse);
+                $this->_sort_filelist($children, $callback, $reverse, $groupbyextflag);
                 $files[$i]['children'] = $children;
             }
         }
 
-        // sort current tree
-        usort($files, $callback);
-        if ($reverse) {
-            $files = array_reverse($files);
+        // sort files by ext
+        $filesByExt = array();
+        foreach($files as $file) {
+            $filesByExt[$file['ext']][] = $file;
+        }
+
+        if ($groupbyextflag) {  // sort for each extension
+            $sortedFiles = array();
+            foreach ($filesByExt as $ext=>$file) {
+                usort($file, $callback);
+                if ($reverse) {
+                    $file = array_reverse($file);
+                }
+                //merge sorted files
+                $sortedFiles = array_merge($sortedFiles, $file);
+            }
+            $files = $sortedFiles;
+        } else {
+            // sort all files
+            usort($files, $callback);
+            if ($reverse) {
+                $files = array_reverse($files);
+            }
         }
     }
 
@@ -618,10 +683,13 @@
                         }
                     }
 
+                    // extension
+                    $fileext =  end(explode(".", $filename));
                     // prepare entry
                     if (!is_dir($filepath) || $params['recursive']) {
                         $entry = array(
                             'name' => $filename,
+                            'ext' => $fileext,
                             'path' => $filepath,
                             'mtime' => filemtime($filepath),
                             'ctime' => filectime($filepath),
@@ -661,6 +729,7 @@
         return strcmp(strtolower($a['name']), strtolower($b['name']));
     }
 
+
     function _compare_ctimes($a, $b) {
         if ($a['ctime'] == $b['ctime']) {
             return 0;
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/de/lang.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/de/lang.php	2010-12-15 12:23:55.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/de/lang.php	2010-12-15 18:56:25.000000000 +0100
@@ -12,3 +12,5 @@
 $lang['lastmodified'] = 'Letzte Änderung';
 $lang['error_nomatch'] = 'Keine Treffer';
 $lang['error_outsidejail'] = 'Zugriff verweigert';
+$lang['edit_files'] = "Dateien bearbeiten";
+$lang['refresh_filelist'] = "Liste aktualisieren";
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/en/lang.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/en/lang.php	2010-12-15 12:23:55.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/lang/en/lang.php	2010-12-15 18:56:25.000000000 +0100
@@ -12,3 +12,5 @@
 $lang['lastmodified'] = 'Last modified';
 $lang['error_nomatch'] = 'No match';
 $lang['error_outsidejail'] = 'Access denied';
+$lang['edit_files'] = "Edit files";
+$lang['refresh_filelist'] = "Refresh list";
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/style.css
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/style.css	2010-12-15 12:23:55.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/style.css	2010-12-15 18:56:25.000000000 +0100
@@ -1,7 +1,17 @@
-a.fleditfiles {
-    display:block;
-    padding: 2px 0 2px 20px;
+div.filelist-toolbar {
+    margin-bottom: 3px;
+}
+
+div.filelist-toolbar a.filelist-editfiles {
+    padding: 2px 0 4px 20px;
+    margin-right: 8px;
     text-align: left;
-    color: #454545;
+    color: #454545 !important;
     background: transparent url("img/folder.png") top left no-repeat;
 }
+a.filelist-refresh {
+    padding: 2px 0 2px 20px;
+    text-align: left;
+    color: #454545 !important;
+    background: transparent url("img/reload.png") top left no-repeat;
+}
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/default.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/default.php	2010-12-15 12:23:56.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/default.php	2010-12-15 18:56:25.000000000 +0100
@@ -6,3 +6,4 @@
 $conf['allow_in_comments'] = 0;
 $conf['allowed_absolute_paths'] = DOKU_INC;
 $conf['web_paths'] = DOKU_URL;
+$conf['simplefilelist_defaults'] = 'tableheader,tableshowdate,tableshowsize,toolbar';
Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/metadata.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/metadata.php	2010-12-15 12:23:56.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/filelist/conf/metadata.php	2010-12-15 18:56:25.000000000 +0100
@@ -9,3 +9,4 @@
 $meta['allow_in_comments'] = array('onoff');
 $meta['allowed_absolute_paths'] = array('');
 $meta['web_paths'] = array('');
+$meta['simplefilelist_defaults'] = array('multicheckbox','_choices' => array('tableheader','tableshowdate','tableshowsize','recursive','toolbar','groupbyext'));
