Index: openschulportfolio/usr/share/linuxmuster-portfolio/lib/plugins/edittable/action.php
===================================================================
--- openschulportfolio.orig/usr/share/linuxmuster-portfolio/lib/plugins/edittable/action.php	2012-12-16 19:42:43.000000000 +0100
+++ openschulportfolio/usr/share/linuxmuster-portfolio/lib/plugins/edittable/action.php	2012-12-16 19:44:11.000000000 +0100
@@ -153,11 +153,13 @@
 
         global $TEXT;
         if (isset($_POST['do']['edit'])) {
-            $_REQUEST['target'] = 'table';
-            $TEXT = "^  ^  ^\n";
+            $_REQUEST['target'] = 'section';
+            $TEXT = $_POST['edittable__new']['pre'] . "\n";
+            $TEXT .= "^  ^  ^\n";
             foreach (explode("\n", $_POST['edittable__new']['text']) as $line) {
                 $TEXT .= "| $line |  |\n";
             }
+            $TEXT .= $_POST['edittable__new']['suf'] . "\n";
         } elseif (isset($_POST['do']['draftdel'])) {
             $TEXT = $_POST['edittable__new']['pre'] .
                     $_POST['edittable__new']['text'] .
