Index: trunk/usr/share/linuxmuster-portfolio/inc/parser/xhtml.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/inc/parser/xhtml.php	2010-10-17 19:03:13.000000000 +0200
+++ trunk/usr/share/linuxmuster-portfolio/inc/parser/xhtml.php	2010-10-17 19:03:33.000000000 +0200
@@ -902,14 +902,14 @@
     }
 
     // $numrows not yet implemented
-    function table_open($maxcols = NULL, $numrows = NULL, $pos){
+    function table_open($maxcols = NULL, $numrows = NULL, $pos=NULL){
         global $lang;
         // initialize the row counter used for classes
         $this->_counter['row_counter'] = 0;
         $this->doc .= '<div class="table ' . $this->startSectionEdit($pos, 'table') . '"><table class="inline">'.DOKU_LF;
     }
 
-    function table_close($pos){
+    function table_close($pos=NULL){
         $this->doc .= '</table></div>'.DOKU_LF;
         $this->finishSectionEdit($pos);
     }
