Index: trunk/usr/share/linuxmuster-portfolio/lib/plugins/task/action.php
===================================================================
--- trunk.orig/usr/share/linuxmuster-portfolio/lib/plugins/task/action.php	2012-01-31 13:19:09.000000000 +0100
+++ trunk/usr/share/linuxmuster-portfolio/lib/plugins/task/action.php	2012-01-31 13:20:55.000000000 +0100
@@ -101,6 +101,14 @@
                         );
 
                 $TEXT = $this->_pageTemplate($data);
+                // --- begin of hack ---
+                $meta = array(
+                        'user'   => $user,
+                        'date'   => $date,
+                        'status' => 0
+                );
+                p_set_metadata($ID, array('task' => $meta));
+                // --- end of hack ---
                 return 'preview';
             } else {
                 return 'edit';
